webui/src/pages/repositories/repository/fileRenderers/editor.jsx

```diff
@@ -0,0 +1,42 @@
+import React, {useCallback, useState} from 'react';
+import Editor  from 'react-simple-code-editor';
+import Prism from 'prismjs';
+import 'prismjs/components/prism-sql';
+import "../../../../styles/ghsyntax.css";
+
+
+export const SQLEditor = ({ initialValue, onChange, onRun }) => {
+    const isMac = window.navigator.platform.toLowerCase().startsWith('mac');
```

Comment:
`navigator.platform` is deprecated ([ref](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform))

I also think you can streamline this quite a bit. The key down event object has properties for modifiers
`ctrlKey`, `shiftKey`, `altKey`, and `metaKey`. They are booleans indicating if the respective modifiers are pressed along with the key which triggered the event.
I believe `ctrlKey` abstracts away the need for detecting Mac vs. PC, since it registered for both `Ctrl` and `Cmd`. If you're still interested in the Meta key, you can add an `||` to the if.
```JavaScript
if (e.key === "Enter" && (e.ctrlKey || e.metaKey))`
```

Reply:
nice! was not aware. Implemented and simplified that code significantly :) thanks